### PR TITLE
Binary data binding

### LINKED ISSUE #17473: [WebJobs/EventHubs] Add support for BinaryData binding
Allow BinaryData parameters to be used in WebJob function invocations.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/17473

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs [hunk 1]
```diff
@@ -90,6 +90,27 @@ public async Task EventHub_SingleDispatch()
...
+        }
+
+        [Test]
+        public async Task EventHub_SingleDispatch_BinaryData()
+        {
+            var (jobHost, host) = BuildHost<EventHubTestSingleDispatchJobsBinaryData>();
+            using (jobHost)
+            {
+                var method = typeof(EventHubTestSingleDispatchJobsBinaryData).GetMethod(nameof(EventHubTestSingleDispatchJobsBinaryData.SendEvent_TestHub), BindingFlags.Static | BindingFlags.Public);
+                await jobHost.CallAsync(method, new { input = _testId });
```
Comment (pakrym) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs:103]:
```suggestion
                await jobHost.CallAsync(nameof(EventHubTestSingleDispatchJobsBinaryData.SendEvent_TestHub),, new { input = _testId });
```

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs [hunk 2]
```diff
@@ -312,6 +381,47 @@ public static void SendEvents_TestHub(int numEvents, string input, [EventHub(Tes
             }
         }
 
+        public class EventHubTestMultipleDispatchJobsBinaryData
+        {
+            private static int s_eventCount;
+            private static int s_processedEventCount;
+            public static void SendEvents_TestHub(int numEvents, string input, [EventHub(TestHubName)] out EventData[] events)
```
Comment (pakrym) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs:388]: Let's use BinaryData also for sending in both tests.

**Review by pakrym — APPROVED** [review]

**Pull request merged**